### PR TITLE
[IMP] sales_team: display proper names in graph legend(s) in sale's dashboard

### PR DESCRIPTION
**Pull request by mitali-willdooit**
Description of the issue/feature this PR addresses:
<b>Task:</b> https://www.odoo.com/web#id=31288&view_type=form&model=project.task&action=333&active_id=131&menu_id=4720
<b>Pad:</b> https://pad.odoo.com/p/r.c6c0d3f37349f79a8f8aecf0fd65a282
Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/sales_team/static/src/js/sales_team_dashboard_widget.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
+odoo.define('sales_team.dashboard_widgets', function (require) {
+"use strict";
+
+var KanbanWidgets = require('web_kanban.widgets');
+
+KanbanWidgets.registry.get('dashboard_graph').include({
```
Comment (tde-banana-odoo) [inline addons/sales_team/static/src/js/sales_team_dashboard_widget.js]: melanchon_gros_yeux

File: addons/web/static/tests/fields/basic_fields_tests.js [hunk 2]
```diff
@@ -864,8 +864,9 @@ QUnit.module('basic_fields', {
 
     QUnit.test('graph dashboard widget is rendered correctly', function (assert) {
         var done = assert.async();
-        assert.expect(3);
+        assert.expect(4);
```
Comment (tde-banana-odoo) [inline addons/web/static/tests/fields/basic_fields_tests.js]: Mmh ?

File: addons/web/static/tests/fields/basic_fields_tests.js [hunk 3]
```diff
@@ -888,6 +889,7 @@ QUnit.module('basic_fields', {
             key: 'A key',
             area: true,
         }]);
+        graph_key = JSON.parse(this.data.partner.records[0].graph_data)[0].key
```
Comment (tde-banana-odoo) [inline addons/web/static/tests/fields/basic_fields_tests.js]: Why defining the variable at line 869 and setting it here ?

File: addons/account/models/account_journal_dashboard.py [hunk 4]
```diff
@@ -87,7 +98,9 @@ def get_line_graph_datas(self):
                 short_name = format_date(show_date, 'd MMM', locale=locale)
                 data.append({'x': short_name, 'y':last_balance, 'name': name})
 
-        return [{'values': data, 'area': True}]
+        [graph_title, graph_key] = self._graph_title_and_key()
+        color = '#875A7B' if '+e' in version else '#7c7bad'
```
Comment (tde-banana-odoo) [inline addons/account/models/account_journal_dashboard.py]: Whut ?

File: addons/web/static/tests/fields/basic_fields_tests.js (hunk 2 above)
Comment (Icallhimtest) [inline addons/web/static/tests/fields/basic_fields_tests.js]: the number of assertions that should be made during the test.

Basicly this attempts to ensure everything that needs to be tested is tested, and that things aren't rendered twice or things like that.

File: addons/web/static/tests/fields/basic_fields_tests.js (hunk 3 above)
Comment (Icallhimtest) [inline addons/web/static/tests/fields/basic_fields_tests.js]: Guess it's cause the key is fetched just before.

File: addons/account/models/account_journal_dashboard.py (hunk 4 above)
Comment (Icallhimtest) [inline addons/account/models/account_journal_dashboard.py]: ugly hack to get the primary color in enterprise/community.

I had asked qsm how it should be fetched and he said there was no clean way. That there were plenty of places where it's hard-coded.

File: addons/web/static/tests/fields/basic_fields_tests.js (hunk 2 above)
Comment (tde-banana-odoo) [inline addons/web/static/tests/fields/basic_fields_tests.js]: Strange.

File: addons/web/static/tests/fields/basic_fields_tests.js (hunk 3 above)
Comment (tde-banana-odoo) [inline addons/web/static/tests/fields/basic_fields_tests.js]: Well I suppose you could directly asign graph_key here.

**Comment by Icallhimtest** [comment]
landed in master @ https://github.com/odoo/odoo/commit/1314de50989901f9ea290b19676a6a6fa15496f8

**Pull request merged**